core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/DistanceRangeMapImpl.kt

```diff
@@ -33,7 +33,36 @@ data class DistanceRangeMapImpl<T>(
      * structure (RangeMaps lib).
      */
     override fun putMany(entries: List<DistanceRangeMap.RangeMapEntry<T>>) {
+        // Unfortunately, built-in groupBy doesn't consider the original order, here we want
+        // consecutive values.
+        fun <U, V> groupByFirst(pairs: List<Pair<U, V>>): List<Pair<U, List<V>>> {
```

Comment:
Can you make it clear that this variant of groupBy only groups consecutive values?